File: environment/src/main/java/com/sequenceiq/environment/exception/mapper/AccessDeniedExceptionMapper.java
package com.sequenceiq.environment.exception.mapper;

import javax.ws.rs.core.Response.Status;

import org.springframework.security.access.AccessDeniedException;
import org.springframework.stereotype.Component;

@Component
public class AccessDeniedExceptionMapper extends EnvironmentBaseExceptionMapper<AccessDeniedException> {

    @Override
    public Status getResponseStatus(AccessDeniedException exception) {
        return Status.FORBIDDEN;
    }

    @Override
    public Class<AccessDeniedException> getExceptionType() {
        return AccessDeniedException.class;
    }
}

File: autoscale/src/main/java/com/sequenceiq/periscope/rest/mapper/AccessDeniedExceptionMapper.kt
package com.sequenceiq.periscope.rest.mapper

import java.nio.file.AccessDeniedException

import javax.ws.rs.core.Response
import javax.ws.rs.ext.ExceptionMapper
import javax.ws.rs.ext.Provider

import org.slf4j.Logger
import org.slf4j.LoggerFactory

import com.sequenceiq.periscope.api.model.ExceptionResult

@Provider
class AccessDeniedExceptionMapper : ExceptionMapper<AccessDeniedException> {

    override fun toResponse(exception: AccessDeniedException): Response {
        LOGGER.error(exception.message, exception)
        return Response.status(Response.Status.FORBIDDEN).entity(ExceptionResult(exception.message)).build()
    }

    companion object {

        private val LOGGER = LoggerFactory.getLogger(AccessDeniedExceptionMapper::class.java)
    }
}
